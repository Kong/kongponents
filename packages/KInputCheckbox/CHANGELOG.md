# @kongponents/kinputcheckbox

## Table of Contents

- [0.1.1](#011)

### 0.1.1
 - :rocket: initial release
