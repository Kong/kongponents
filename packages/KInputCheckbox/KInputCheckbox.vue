<template>
  <label class="k-inputCheckbox">
    <input
      :checked="value"
      v-bind="$attrs"
      type="checkbox"
      v-on="listeners">
    <slot>{{ label }}</slot>
  </label>
</template>

<script>
export default {
  name: 'KInputCheckbox',
  props: {
    /**
     * Sets whether or not checkbox is checked
     */
    value: {
      type: Boolean,
      default: false,
      required: true
    },

    /**
     * Sets label text
     */
    label: {
      type: String,
      default: ''
    }
  },

  computed: {
    listeners () {
      const listeners = {
        ...this.$listeners,
        change: e => {
          this.$emit('change', e.target.checked)
          this.$emit('input', e.target.checked)
        }
      }

      // remove 'input' since it doesn't need to be bound to the checkbox
      delete listeners['input']

      return listeners
    }
  }
}
</script>

<style lang="scss" scoped>
@import '~@kongponents/styles/_variables.scss';
$primaryColor: var(--KInputCheckboxPrimary, var(--blue-base, color(blue-base)));
$borderColor: var(--KInputCheckboxBorder, var(--tblack-25, color(tblack-25)));

.k-inputCheckbox {
  input[type="checkbox"] {
    display: inline-block;
    vertical-align: middle;
    height: 20px;
    width: 20px;
    padding: 0;
    color: $primaryColor;
    border: 1px solid $borderColor;
    border-radius: 0.25rem;
    box-sizing: border-box;
    background-color: #fff;

    // Overide browser defaults
    -webkit-appearance: none;
       -moz-appearance: none;
            appearance: none;
    -webkit-print-color-adjust: exact;
            user-select: none;
    -webkit-user-select: none;
       -moz-user-select: none;
        -ms-user-select: none;

    &:checked {
      background-image: url("data:image/svg+xml,%3Csvg width='12' height='10' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M10.633 0L12 1.397 3.583 10 0 6.337 1.367 4.94l2.216 2.265z' fill='%23FFF' fill-rule='nonzero'/%3E%3C/svg%3E");
      border-color: transparent;
      background-color: currentColor;
      background-position: center;
      background-repeat: no-repeat;
    }

    &:hover {
      border-color: $primaryColor;
    }

    &:focus {
      outline: none;
      border-color: $primaryColor;
    }

    &:disabled {
      background-color: var(--KInputCheckboxDisabled, var(--grey-92, color(grey-92)));
      cursor: not-allowed;
      &:hover {
        border-color: $borderColor;
      }
    }
  }
}
</style>
