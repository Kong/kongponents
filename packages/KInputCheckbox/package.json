{
  "name": "@kongponents/kinputcheckbox",
  "version": "0.1.0",
  "description": "KInputCheckbox description here.",
  "main": "KInputCheckbox.vue",
  "scripts": {
    "test": "jest"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "homepage": "https://github.com/Kong/kongponents#readme"
}
