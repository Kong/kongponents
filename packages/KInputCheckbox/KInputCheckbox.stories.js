import { storiesOf } from '../../storybook/node_modules/@storybook/vue/dist/client'
import KInputCheckbox from './KInputCheckbox.vue'

storiesOf('KInputCheckbox', module)
  .add('Default', () => ({
    components: { KInputCheckbox },
    template: `
      <div>
        <KInputCheckbox
          v-model="checked"
          @change="handleToggle">
          {{ labelText }}
        </KInputCheckbox>
        <KInputCheckbox disabled>
          disabled
        </KInputCheckbox>
    `,
    data: () => ({
      checked: false
    }),
    computed: {
      labelText () {
        return this.checked ? 'on' : 'off'
      }
    },
    methods: {
      handleToggle (value) {
        this.checked = value
        console.log(this.checked)
      }
    }
  }), { info: {} })
