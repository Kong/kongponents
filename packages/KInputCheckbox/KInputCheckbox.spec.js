import { mount } from '@vue/test-utils'
import KInputCheckbox from '@/KInputCheckbox/KInputCheckbox'

describe('KInputCheckbox', () => {
  it('shows as checked when prop passed', () => {
    const wrapper = mount(KInputCheckbox, {
      propsData: {
        value: true
      }
    })

    expect(wrapper.find('input').element.checked).toBe(true)
  })

  it('emits checked value on click', () => {
    const wrapper = mount(KInputCheckbox, {
      propsData: {
        value: true
      }
    })
    const toggle = wrapper.find('input')

    toggle.trigger('click')

    expect(wrapper.emitted().change).toBeTruthy()
    expect(wrapper.emitted().change).toHaveLength(1)
  })

  it('matches snapshot', () => {
    const wrapper = mount(KInputCheckbox, {
      propsData: {
        value: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
