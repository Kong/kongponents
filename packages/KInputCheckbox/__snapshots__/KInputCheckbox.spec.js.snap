// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KInputCheckbox matches snapshot 1`] = `<label class="k-inputCheckbox"><input type="checkbox"> </label>`;
